addons/point_of_sale/static/src/app/store/db.js

```diff
@@ -567,18 +592,33 @@ export class PosDB {
         } catch {
             return [];
         }
-        var results = [];
-        for (var i = 0; i < this.limit; i++) {
-            var r = re.exec(this.category_search_string[category_id]);
-            if (r) {
-                var id = Number(r[1]);
-                const product = this.get_product_by_id(id);
-                if (!this.shouldAddProduct(product, results)) {
-                    continue;
+        let results = [];
+        if(filteredProducts.length > 0){
+            for(let j = 0; j < filteredProducts.length;j++){
+                let r = re.exec(filteredProducts[j].id + ":" + filteredProducts[j].display_name);
+                if (r) {
+                    var id = Number(r[1]);
```

Comment:
`var` is super old school and should not be used (except if really necessary). Depending on the context, prefer the usage of `const` or `let`